source/ios/tools/IOSFeed/IOSFeed.cs

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.Blob;
+
+namespace AdaptiveCards.Tools.IOSFeed
+{
+    public static class Constants
+    {
+        public const string ConnectionStringPath = "source/ios/tools/IOSFeed/ConnectString.txt";
+        public const string ContainerId = "adaptivecardsiosblobs";
+        public const string FrameworkPath = "source/ios/AdaptiveCards/AdaptiveCards";
+        public const string FrameworkName = "AdaptiveCards.framework.zip";
+        public const string PodspecPath = "./source/ios/tools/";
+        public const string TargetPodspecPath = "./source/ios/";
+        public const string PodspecName = "AdaptiveCards.podspec";
+    }
+
+    /// <summary>
+    /// IOSFeed picks up ios artifact, sends to the Azure storage 
+    /// and updates podspec for pod publication
+    /// </summary>
+    class IOSFeed
+    {
+        public static void Main()
+        {
+            // Run the examples asynchronously, wait for the results before proceeding
+            ProcessAsync().GetAwaiter().GetResult();
+        }
+
+        private static async Task ProcessAsync()
+        {
+            string storageConnectionString;
+
+            FileInfo fi = new FileInfo(Constants.ConnectionStringPath);
+            if (!fi.Exists)
+            {
+                throw new InvalidOperationException("connection string file does not exist");
+            }
+
+            using (StreamReader sr = File.OpenText(Constants.ConnectionStringPath))
+            {
+                storageConnectionString = sr.ReadToEnd();
+            }
+
+            // Check whether the connection string can be parsed.
+            CloudStorageAccount storageAccount;
+            if (CloudStorageAccount.TryParse(storageConnectionString, out storageAccount))
+            {
+                // If the connection string is valid, proceed with operations against Blob
+                // storage here.
+                // Create the CloudBlobClient that represents the 
+                // Blob storage endpoint for the storage account.
+                CloudBlobClient cloudBlobClient = storageAccount.CreateCloudBlobClient();
+
+                CloudBlobContainer cloudBlobContainer =
+                    cloudBlobClient.GetContainerReference(Constants.ContainerId);
+                await cloudBlobContainer.CreateIfNotExistsAsync();
+
+                // Set the permissions so the blobs are public.
+                BlobContainerPermissions permissions = new BlobContainerPermissions
+                {
+                    PublicAccess = BlobContainerPublicAccessType.Blob
+                };
+
```

Comment:
What happens if any of the code fails below -- do you want to delete the blob you created so we dont have orphans laying around? Suggest add a try-catch-finally block

Reply:
sound great; thanks

---
In reply to: [314079529](https://github.com/microsoft/AdaptiveCards/pull/3291#discussion_r314079529) [](ancestors = 314079529)